app/models/document.rb

```diff
@@ -173,6 +169,10 @@ def self.extract_body_from_payload(payload)
     end
   end
 
+  def self.extract_update_type_from_payload(payload)
+    payload['update_type'] if payload['publication_state'] == 'redrafted'
+  end
```

Comment:
so I'm not certain on this, but it -seems- that that doesn't work, because adding an attachment needs an update type. If you add an attachment to a non-redrafted, non-draft document then the update type gets set as nil and everything gets sad. 

Not really sure how to combine them; as far as I can tell one is a change that can only happen on the way up to the view, while the other has to happen both to the view and the db.